---
pageClass: "rule-details"
sidebarDepth: 0
title: "regexp/no-useless-character-class"
description: "disallow character class with one character"
since: "v0.3.0"
---
# regexp/no-useless-character-class

ðŸ’¼ This rule is enabled in the âœ… `plugin:regexp/recommended` config.

ðŸ”§ This rule is automatically fixable by the [`--fix` CLI option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix).

<!-- end auto-generated rule header -->

> disallow character class with one character

## :book: Rule Details

This rule reports character classes that defines only one character.

Character classes that define only one character have the same effect even if you remove the brackets.

<eslint-code-block fix>

```js
/* eslint regexp/no-useless-character-class: "error" */

/* âœ“ GOOD */
var foo = /abc/;

/* âœ— BAD */
var foo = /a[b]c/;
```

</eslint-code-block>

## :wrench: Options

```json
{
  "regexp/no-useless-character-class": ["error", {
    "ignores": ["="]
  }]
}
```

- `"ignores"` ... An array of characters and character classes to ignores. Default `["="]`.

The default value is `"="` to prevent conflicts with the [no-div-regex] rule. Note that if you do not specify `"="`, there may be conflicts with the [no-div-regex] rule.

### `"ignores": ["a"]`

<eslint-code-block fix>

```js
/* eslint regexp/no-useless-character-class: ["error", { "ignores": ["a"] }] */

/* âœ“ GOOD */
var foo = /[a]bc/;

/* âœ— BAD */
var foo = /a[b]c/;
```

</eslint-code-block>

## :couple: Related rules

- [no-empty-character-class]
- [no-div-regex]

[no-empty-character-class]: https://eslint.org/docs/rules/no-empty-character-class
[no-div-regex]: https://eslint.org/docs/rules/no-div-regex

## :rocket: Version

This rule was introduced in eslint-plugin-regexp v0.3.0

## :mag: Implementation

- [Rule source](https://github.com/ota-meshi/eslint-plugin-regexp/blob/master/lib/rules/no-useless-character-class.ts)
- [Test source](https://github.com/ota-meshi/eslint-plugin-regexp/blob/master/tests/lib/rules/no-useless-character-class.ts)
