---
pageClass: "rule-details"
sidebarDepth: 0
title: "regexp/sort-alternatives"
description: "sort alternatives if order doesn't matter"
---
# regexp/sort-alternatives

> sort alternatives if order doesn't matter

- :exclamation: <badge text="This rule has not been released yet." vertical="middle" type="error"> ***This rule has not been released yet.*** </badge>
- :wrench: The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.

## :book: Rule Details

This rule reports ???.

<eslint-code-block fix>

```js
/* eslint regexp/sort-alternatives: "error" */

/* ✓ GOOD */


/* ✗ BAD */

```

</eslint-code-block>

## :wrench: Options

Nothing.

## :mag: Implementation

- [Rule source](https://github.com/ota-meshi/eslint-plugin-regexp/blob/master/lib/rules/sort-alternatives.ts)
- [Test source](https://github.com/ota-meshi/eslint-plugin-regexp/blob/master/tests/lib/rules/sort-alternatives.ts)
